### PR TITLE
Use absl::SkipEmpty() predicate.

### PR DESCRIPTION
**Pull request by MichaelGrupp**
Fixes empty splits for default "" arguments. Follow up to #1026, thx to @ojura.

### CODE REVIEW

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Error: PUT https://api.github.com/repos/googlecartographer/cartographer_ros/pulls/1042/merge: 405 At least 1 approving review is required by reviewers with write access. []

Log:
~~~
=> handle work item: PR 1042 on googlecartographer/cartographer_ros.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer_ros]
HEAD is now at 8fab3fe Work around wstool bug in build with Docker base image. (#1041)
 $ git clean -fd [data/cartographer_ros]
 $ git checkout master [data/cartographer_ros]
Already on 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer_ros]
Already up to date.
 => Checking out magazino/skip-whitespace as local branch pr_1042.
 $ git remote add magazino git@github.com:magazino/cartographer_ros.git [data/cartographer_ros]
fatal: remote magazino already exists.
 $ git fetch magazino [data/cartographer_ros]
 $ git branch --track pr_1042 magazino/skip-whitespace [data/cartographer_ros]
Branch 'pr_1042' set up to track remote branch 'skip-whitespace' from 'magazino'.
 => Switching to branch pr_1042.
 $ git reset --hard HEAD [data/cartographer_ros]
HEAD is now at 8fab3fe Work around wstool bug in build with Docker base image. (#1041)
 $ git clean -fd [data/cartographer_ros]
 $ git checkout pr_1042 [data/cartographer_ros]
Switched to branch 'pr_1042'
Your branch is up to date with 'magazino/skip-whitespace'.
 $ git pull --force [data/cartographer_ros]
Already up to date.
 => merge master.
 $ git merge master [data/cartographer_ros]
Already up to date.
 => Running clang-format on all *.cc and *.h files
 $ git commit -am Ran clang-format. [data/cartographer_ros]
On branch pr_1042
Your branch is up to date with 'magazino/skip-whitespace'.

nothing to commit, working tree clean
 $ git push --force magazino HEAD:skip-whitespace [data/cartographer_ros]
Everything up-to-date
 => Abandoning branch pr_1042.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer_ros]
HEAD is now at a30cc46 Merge branch 'master' into skip-whitespace
 $ git clean -fd [data/cartographer_ros]
 $ git checkout master [data/cartographer_ros]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
 $ git branch -D pr_1042 [data/cartographer_ros]
Deleted branch pr_1042 (was a30cc46).

~~~

File: cartographer_ros/cartographer_ros/assets_writer_main.cc [hunk 1]
```diff
@@ -55,7 +55,8 @@ int main(int argc, char** argv) {
       << "-pose_graph_filename is missing.";
 
   ::cartographer_ros::AssetsWriter asset_writer(
-      FLAGS_pose_graph_filename, absl::StrSplit(FLAGS_bag_filenames, ','),
+      FLAGS_pose_graph_filename,
+      absl::StrSplit(FLAGS_bag_filenames, ',', absl::SkipWhitespace()),
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/assets_writer_main.cc]: I think it is much better to use `absl::SkipEmpty()` here. `" "` is a valid filename for which this tool now no longer works correctly, i.e. I would consider this a bug.

Comment (MichaelGrupp) [reply cartographer_ros/cartographer_ros/assets_writer_main.cc]: Ok. Let's change it before Wally wakes up from the dead.

Comment (MichaelGrupp) [reply cartographer_ros/cartographer_ros/assets_writer_main.cc]: Done.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.

**Pull request merged**